tests/fixed-foveation.html

```diff
@@ -0,0 +1,553 @@
+          0, 4, 5, 5, 3, 0,	// left
+          3, 5, 6, 6, 2, 3,	// front
+          0, 1, 7, 7, 4, 0	// back
+        ];
+
+        cube.IndexBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, cube.IndexBuffer);
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array(cubeIndices), gl.STATIC_DRAW);
+        console.log("cube.IndexBuffer size " + gl.getBufferParameter(gl.ELEMENT_ARRAY_BUFFER, gl.BUFFER_SIZE));
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, null);
+
+        return cube;
+      }
+
+      function GeometryCreateVAO(gl, geometry) {
+        geometry.VertexArrayObject = gl.createVertexArray();
+        gl.bindVertexArray(geometry.VertexArrayObject);
+        gl.bindVertexArray(null);
+      }
+
+      function SceneCreateVAOs(gl, scene)
+      {
+        if (!scene.CreatedVAOs )
+        {
+          GeometryCreateVAO(gl, scene.Cube);
+
+          // Modify the VAO to use the instance transform attributes.
+          gl.bindVertexArray(scene.Cube.VertexArrayObject);
+          gl.bindBuffer(gl.ARRAY_BUFFER, scene.InstanceTransformBuffer);
+          gl.bindVertexArray(null);
+
+          scene.CreatedVAOs = true;
+        }
+      }
+
+      function createScene(gl)
+      {
+        //ovrProgram_Create( &scene->Program, VERTEX_SHADER, FRAGMENT_SHADER, useMultiview );
+        var obj = new Object;
+
+        obj.Cube = createCube(gl);
+        //var instArr = new Float32Array(NUM_INSTANCES*4*4);
+        // Create the instance transform attribute buffer.
+        obj.InstanceTransformBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, obj.InstanceTransformBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, NUM_INSTANCES * 4 * 4 * 4, gl.DYNAMIC_DRAW);
+        console.log("InstanceTransformBuffer size " + gl.getBufferParameter(gl.ARRAY_BUFFER, gl.BUFFER_SIZE));
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+        obj.SceneMatrices = gl.createBuffer();
+        gl.bindBuffer(gl.UNIFORM_BUFFER, obj.SceneMatrices);
+        gl.bufferData(gl.UNIFORM_BUFFER, 2 * 16 * 4 /*sizeof( ovrMatrix4f )*/ /* 2 view matrices */ + 2 * 16 * 4 /*sizeof( ovrMatrix4f )*/ /* 2 projection matrices */, gl.STATIC_DRAW);
+        gl.bindBuffer(gl.UNIFORM_BUFFER, null);
+
+        // Setup random rotations.
+        obj.Rotations = new Array(NUM_ROTATIONS);
+        for ( var i = 0; i < NUM_ROTATIONS; i++ )
+        {
+          obj.Rotations[i] = new Object;
+          obj.Rotations[i].x = Math.random();
+          obj.Rotations[i].y = Math.random();
+          obj.Rotations[i].z = Math.random();
+        }
+
+        // Setup random cube positions and rotations.
+        obj.CubePositions = new Array(NUM_INSTANCES);
+        obj.CubeRotations = new Int32Array(NUM_INSTANCES);
+        for ( var i = 0; i < NUM_INSTANCES; i++ )
+        {
+          obj.CubePositions[i] = { x:0, y:0, z:0 };
+          // Using volatile keeps the compiler from optimizing away multiple calls to ovrScene_RandomFloat().
```

Comment:
It would be worth doing a pass over the comments to make sure they still make sense in this context. `volatile` isn't  a concept that your average JavaScript developer knows about (or wants to know about, for that matter...😁)